File: domain/categories-domain/src/main/java/com/fit/fitndflow/app/domain/categories/repository/CategoriesRepository.java
package com.fit.fitndflow.app.domain.categories.repository;

import com.fit.fitndflow.app.domain.common.models.CategoryModel;
import com.fit.fitndflow.app.domain.common.repository.CommonRepository;
import java.util.List;

public interface CategoriesRepository extends CommonRepository {
    List<CategoryModel> getCategoryList() throws Exception;
    List<CategoryModel> addNewCategory(String categoryName, String language) throws Exception;
    List<CategoryModel> modifyCategory(String categoryName, String language, int categoryId, String imageUrl) throws Exception;
    List<CategoryModel> deleteCategory(Integer integer) throws Exception;
}

File: domain/categories-domain/src/main/java/com/fit/fitndflow/app/domain/categories/repository/CategoriesRepository.kt
package com.fit.fitndflow.app.domain.categories.repository

import com.fit.fitndflow.app.domain.common.models.CategoryModel
import com.fit.fitndflow.app.domain.common.repository.CommonRepository

interface CategoriesRepository : CommonRepository {
    suspend fun getCategoryList(): List<CategoryModel>
    suspend fun addNewCategory(categoryName: String, language: String): List<CategoryModel>
    suspend fun modifyCategory(
        categoryName: String,
        language: String,
        categoryId: Int,
        imageUrl: String?
    ): List<CategoryModel>
    suspend fun deleteCategory(integer: Int): List<CategoryModel>

    suspend fun migrateToLocal()

    suspend fun createInitialData()

    fun isInitialDataCreated(): Boolean
}
